File: src/main/java/com/vaadin/demo/component/horizontallayout/HorizontalLayoutSlotsWrapping.java
package com.vaadin.demo.component.horizontallayout;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.Route;

@Route("horizontal-layout-slots-wrapping")
public class HorizontalLayoutSlotsWrapping extends Div {
    public HorizontalLayoutSlotsWrapping() {
        // tag::snippet[]
        HorizontalLayout layout = new HorizontalLayout();
        layout.setWrap(true);
        layout.setPadding(true);

        HorizontalLayout start = new HorizontalLayout();
        start.setPadding(true);

        Div item1 = new Div("Start");
        item1.setClassName("example-item");

        Div item2 = new Div("Start");
        item2.setClassName("example-item");

        start.add(item1, item2);

        HorizontalLayout middle = new HorizontalLayout();
        middle.setPadding(true);

        Div item3 = new Div("Middle");
        item3.setClassName("example-item");

        Div item4 = new Div("Middle");
        item4.setClassName("example-item");

        Div item5 = new Div("Middle");
        item5.setClassName("example-item");

        middle.add(item3, item4, item5);

        HorizontalLayout end = new HorizontalLayout();
        end.setPadding(true);

        Div item6 = new Div("End");
        item6.setClassName("example-item");

        Div item7 = new Div("End");
        item7.setClassName("example-item");

        end.add(item6, item7);

        layout.addToStart(start);
        layout.addToMiddle(middle);
        layout.addToEnd(end);
        // end::snippet[]

        this.setClassName("basic-layouts-example");
        this.add(layout);
    }

    public static class Exporter extends // hidden-source-line
            DemoExporter<HorizontalLayoutSlotsWrapping> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/horizontal-layout/horizontal-layout-slots-wrapping.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/horizontal-layout';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('horizontal-layout-slots-wrapping')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <div class="basic-layouts-example">
        <!-- tag::snippet[] -->
        <vaadin-horizontal-layout theme="spacing padding wrap">
          <vaadin-horizontal-layout theme="spacing padding">
            <div class="example-item">Start</div>
            <div class="example-item">Start</div>
          </vaadin-horizontal-layout>
          <vaadin-horizontal-layout theme="spacing padding" slot="middle">
            <div class="example-item">Middle</div>
            <div class="example-item">Middle</div>
            <div class="example-item">Middle</div>
          </vaadin-horizontal-layout>
          <vaadin-horizontal-layout theme="spacing padding" slot="end">
            <div class="example-item">End</div>
            <div class="example-item">End</div>
          </vaadin-horizontal-layout>
        </vaadin-horizontal-layout>
        <!-- end::snippet[] -->
      </div>
    `;
  }
}
